src/chromo/common.py

```diff
@@ -708,6 +713,30 @@ def kinematics(self, kin):
         self._kinematics = kin
         self._set_kinematics(kin)
 
+    @property
+    def final_state_particles(self):
+        """Returns sorted list"""
+        uids = np.unique(self._final_state_particles)
+        return list(uids[np.argsort(2 * np.abs(uids) - (uids < 0))])
+
+    def _update_final_state_particles(self, pdgid, stable):
+        unique_pdgid = np.unique(pdgid)
+
+        if stable:
+            self._final_state_particles = np.append(
+                self._final_state_particles, unique_pdgid
+            )
+        else:
+            is_stable = np.logical_not(
+                np.isin(self._final_state_particles, unique_pdgid)
+            )
+            self._final_state_particles = self._final_state_particles[is_stable]
+
+        if self._apply_decay_handler:
+            self._decay_handler.set_stable_decaying(
+                theonly_stable_pids=self._final_state_particles
+            )
```

Comment:
I don't see why this new method is needed. You can achieve this with calls to `set_stable`.

Reply:
Yeap, it is possible. Removed it.